examples/bri-3/src/bri/zkp/capabilities/createProof/createProofCommand.handler.ts

```diff
@@ -0,0 +1,29 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { ProofAgent } from '../../agents/proof.agent';
+import { ProofStorageAgent } from '../../agents/proofStorage.agent';
+import { CreateProofCommand } from './createProof.command';
+
+@CommandHandler(CreateProofCommand)
+export class CreateProofCommandHandler
+  implements ICommandHandler<CreateProofCommand>
+{
+  constructor(
+    private agent: ProofAgent,
+    private storageAgent: ProofStorageAgent,
+  ) {}
+
+  async execute(command: CreateProofCommand) {
+    this.agent.throwIfProofInputInvalid(command.document);
+
+    const newProof = this.agent.createNewProof(
+      command.id,
+      null, // TODO: Fetch BpiAccount based on id ,
```

Comment:
this todo should probably be part of agent, but can be fixed before merge i think

Reply:
+1